includes/woopay/class-woopay-session.php

```diff
@@ -583,6 +578,28 @@ private static function create_woopay_nonce( int $uid ) {
 		return substr( wp_hash( $i . '|' . $action . '|' . $uid . '|' . $token, 'nonce' ), -12, 10 );
 	}
 
+	/**
+	 * Checks whether woopay is enabled. This is a duplicate of
+	 * WC_Payments_Features::is_woopay_eligible() except it doesn't rely on WC_Payments being initialized.
+	 *
+	 * @return bool
+	 */
+	private static function is_woopay_eligible() {
+		// Checks for the dependency on Store API AbstractCartRoute.
+		if ( ! class_exists( 'Automattic\WooCommerce\StoreApi\Routes\V1\AbstractCartRoute' ) ) {
+			return false;
+		}
+
+		// read directly from options to avoid loading WC_Payments.
+		$account      = get_option( 'wcpay_account_data', [] );
+		$account_data = [];
+		if ( is_array( $account ) && array_key_exists( 'data', $account ) ) {
+			$account_data = $account['data'];
+		}
+
+		return is_array( $account_data ) && ( $account_data['platform_checkout_eligible'] ?? false );
```

Comment:
Minor nitpick; I think we can just use the null coalescing operator `??` here since [it doesn't throw errors on variables that don't exist](https://www.php.net/manual/en/language.operators.comparison.php#language.operators.comparison.coalesce):

```sh
$ php --interactive
Interactive shell

php > echo $this_does_not_exist['foo']['bar'] ?? 'baz';
baz
php > $text = 'foo';
php > echo $text['foo']['bar'] ?? 'baz';
baz
php > $array = [ 'foo' => [] ];
php > echo $array['foo']['bar'] ?? 'baz';
baz
php > $array = [ 'foo' => [ 'bar' => 'foobar' ] ];
php > echo $array['foo']['bar'] ?? 'baz';
foobar
```

```suggestion
		return get_option( 'wcpay_account_data', [] )['data']['platform_checkout_eligible'] ?? false;
```

Reply:
Thanks so much for this suggestion, was blanking on the best way to simplify this and didn't realize null coalescing was so forgiving.